hbase-shaded-jaeger/pom.xml

```diff
@@ -0,0 +1,176 @@
+        <relativePath>..</relativePath>
+    </parent>
+    <artifactId>hbase-shaded-jaeger</artifactId>
+    <name>Apache HBase Relocated (Shaded) JaegerTracing Libs</name>
+    <description>The shaded uber jar of JaegerTracing library used by HBase.</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.jaegertracing</groupId>
+            <artifactId>jaeger-client</artifactId>
+            <version>${jaegertracing.version}</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <!--Make it so assembly:single does nothing in here-->
+                    <artifactId>maven-assembly-plugin</artifactId>
+                    <configuration>
+                        <skipAssembly>true</skipAssembly>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-shade-plugin</artifactId>
+                    <executions>
+                        <execution>
+                            <id>aggregate-into-a-jar-with-relocated-third-parties</id>
+                            <phase>package</phase>
+                            <goals>
+                                <goal>shade</goal>
+                            </goals>
+                            <configuration>
+                                <createSourcesJar>false</createSourcesJar>
+                                <shadedArtifactAttached>false</shadedArtifactAttached>
+                                <shadeTestJar>false</shadeTestJar>
+
+                                <artifactSet>
+                                    <excludes>
+                                        <exclude>org.apache.httpcomponents:httpclient</exclude>  <!-- hadoop-common has it too -->
+                                        <exclude>commons-logging:commons-logging</exclude>       <!-- transitively added by hbase-common -> commons-validator -->
+                                        <exclude>commons-codec:commons-codec</exclude>           <!-- hbase-common has it too -->
+                                        <exclude>org.apache.httpcomponents:httpcore</exclude>    <!-- hbase-common has it too -->
+                                        <exclude>javax.annotation:javax.annotation-api</exclude> <!-- comes in through JDK -->
+                                        <exclude>org.slf4j:slf4j-api</exclude>                   <!-- hbase-common has it too -->
+                                        <exclude>io.opentracing:*</exclude>                      <!-- hbase-common declares this dependency too -->
+                                    </excludes>
+                                </artifactSet>
+
+                                <relocations>
+                                    <!-- shade gson, which is transitively included via jaeger-core -->
```

Comment:
Probably not worth trying to deduplicate with the shaded gson we use?

Reply:
Thanks @busbey for the suggest. Found a trick to:
1.  exclude gson from the shaded jar, but
2. update the reference to gson classes to the gson in the hbase-shaded-misc jar.

Verified the gson classes are not in the hbase-shaded-jaeger jar and that the byte code references  org.apache.hbase.thirdparty.com.google.gson.*;